### PR TITLE
Changes for adding the functionality for the nsacls library

### PR DESCRIPTION
**Pull request by sanket4373**
- Added the code changes for adding a new function ApplyResource

### CODE REVIEW

File: netscaler/config_test.go [hunk 1]
```diff
@@ -115,6 +116,60 @@ func TestAdd(t *testing.T) {
 	}
 }
 
+func TestApply(t *testing.T) {
+
+	rndIP := randomIP()
```
Comment (chiradeep) [inline netscaler/config_test.go]: AFAIK, Apply only applies to ACLs. Can you rewrite the test to apply ACL?

Comment (sanket4373) [reply netscaler/config_test.go]: I guess that's what the `ApplyResource` function is doing `apply ns acls`, applying the configured changes for acls. I am not sure if there is any such thing as such `apply ns acl` : http://docs.citrix.com/ko-kr/netscaler/11/reference/netscaler-command-reference/ns/ns-acl.html . Correct me if I am missing to read your question.

Comment (chiradeep) [reply netscaler/config_test.go]: What I meant was that TestApply as written tests the creation of lbvserver. It should test apply of ACL.

**Comment by sanket4373** [comment]
@chiradeep: updated the unit test.

**Comment by sanket4373** [comment]
@chiradeep: I hope that unit test is what you were expecting. Kindly let me know if you need any further changes on this one.

**Comment by chiradeep** [comment]
Didn't build. I made some changes and pushed it to https://github.com/chiradeep/go-nitro/tree/sanket4373-feature/nsacls . Please check it out

**Comment by sanket4373** [comment]
What is the error that you are getting?

**Comment by chiradeep** [comment]
# github.com/chiradeep/go-nitro/netscaler
netscaler/config_test.go:27:2: cannot find package "github.com/go-nitro/config/ns" in any of:
	/usr/local/go/src/github.com/go-nitro/config/ns (from $GOROOT)
	/Users/chiradeep/source/go/src/github.com/go-nitro/config/ns (from $GOPATH)

**Comment by sanket4373** [comment]
@chiradeep: I was able to build the SDK using your code changes. It is because you were trying to access the package from this location (the library is yet to be added your repo) `github.com/chiradeep/go-nitro/config/ns` . For testing you will need to replace it with this path: `github.com/go-nitro/config/ns`. That should take care of that error.

**Comment by sanket4373** [comment]
@chiradeep: pushed new changes for the unit tests as per the changes on your branch from here: https://github.com/chiradeep/go-nitro/tree/sanket4373-feature/nsacls

File: netscaler/config_test.go [hunk 2]
```diff
@@ -17,12 +17,14 @@ package netscaler
...
-	"github.com/chiradeep/go-nitro/config/basic"
-	"github.com/chiradeep/go-nitro/config/lb"
 	"log"
 	"math/rand"
 	"testing"
 	"time"
+
+	"github.com/chiradeep/go-nitro/config/basic"
+	"github.com/chiradeep/go-nitro/config/lb"
+	"github.com/go-nitro/config/ns"
```
Comment (chiradeep) [inline netscaler/config_test.go]: Build still fails here. This simply cannot be a valid import. It has to have my repository in it.

Comment (sanket4373) [reply netscaler/config_test.go]: I don't know why it must be failing, only reason i could think of is the path of running it. This is the path from where I am running the code from.
```
$ pwd
~/go/src/github.com/go-nitro
```

Comment (sanket4373) [reply netscaler/config_test.go]: @chiradeep: I was able to build the code by running the code from this path too: `~/go/src/github.com/chiradeep/go-nitro`.  I was going through your PR again and I don't see the config file `config/ns/nsacls.go` in it. https://github.com/chiradeep/go-nitro/commit/42a5d8877682a75ef562bbd4dc50f1fcc7b80b40

Do you mind adding the config file to the library `chiradeep/go-nitro/config/ns/nsacls.go` . Add this:

```
package ns

type Nsacls struct{}
```

Adding this should solve the import problem of `github.com/chiradeep/go-nitro/config/ns` inside `config_test.go`.

Comment (chiradeep) [reply netscaler/config_test.go]: OK, will try.

**Comment by chiradeep** [comment]
Superseded by PR #5

**Pull request closed without merging**